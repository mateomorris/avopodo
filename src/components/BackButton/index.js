import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';

export class BackButton extends React.Component {

    _handlePress = () => {
        this.props.onPress();
    }

    render() {

        return (
            <Text style={[{ 
                color: '#666666', 
                fontWeight: '600', 
                fontSize: 20,
                marginBottom: 10,
                marginLeft: 5,
                marginTop: 10
            }, this.props.style]}>{this.props.text}</Text>
        );

    }
}