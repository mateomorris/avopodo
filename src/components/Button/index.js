import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';
import { MaterialIndicator } from 'react-native-indicators';
import SvgUri from 'react-native-svg-uri';
import tinycolor from 'tinycolor2';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
});

export default class Button extends React.Component {

    _handlePress = () => {
        this.props.onPress();
    }

    render() {

        return (
            <TouchableOpacity style={[{
                paddingLeft: 10,
                paddingRight: 10,
                paddingTop: 5,
                paddingBottom: 5,
                borderColor: 'whitesmoke',
                borderWidth: 2,
                borderRadius: 35,
                flexDirection: 'row'
            }, this.props.style]} onPress={() => {
                this._handlePress()
            }}>
                <Text style={{
                    color: 'white',
                    fontWeight: '900'
                }}>{ this.props.label }</Text>
                <Image style={{
                    height: 20,
                    width: 20,
                    marginLeft: 3
                }} source={this.props.icon} />
            </TouchableOpacity>
        );

    }
}

export class CircleButton extends React.Component {
    _handlePress = () => {
        this.props.onPress();
    }

    render() {

        let { spinner, color, size, icon, style } = this.props

        return (
            <TouchableOpacity style={[
                    style,
                    {justifyContent: 'center', alignItems: 'center', flexDirection: 'column', width: size, height: size }]} onPress={this._handlePress}>
                <MaterialIndicator color={ spinner ? color : 'transparent' } size={size} animationDuration={3000} style={{
                    position: 'absolute',
                }}/>
                <View style={{ 
                    borderRadius: 500,
                    justifyContent: 'center', 
                    alignItems: 'center', 
                    backgroundColor: `${color}7F`,
                    position: 'absolute',
                    padding: size / 10,
                }}>
                    <View style={{
                        height: size, 
                        width: size, 
                        justifyContent: 'center',
                        alignItems: 'center',
                        flex: 1,
                        backgroundColor: color, 
                        borderRadius: 500,
                    }}>
                    {
                        !this.props.testing &&
                        <SvgUri style={{ 
                            justifyContent: 'center', 
                            alignItems: 'center', 
                            borderRadius: 500,
                            zIndex: 0,
                        }} width={size * 0.5} height={size * 0.5} svgXmlData={icon} fill={tinycolor(color).isLight() ? 'black' : 'white'} fillAll={true}/>
                    } 
                    {
                        !icon &&
                        <Text style={{
                            color: tinycolor(color).isLight() ? 'black' : 'white',
                            fontSize: 20,
                            fontWeight: '500'
                        }}>
                            15
                        </Text>
                    }
                    </View>
                </View>
            </TouchableOpacity>
        );

    }
}

CircleButton.defaultProps = {
    color : '#555555',
    spinner : false,
    size : 100,
    style : {}
}