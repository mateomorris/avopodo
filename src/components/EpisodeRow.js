import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity, Alert } from 'react-native';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
    thumbnail: { 
        width: 120, 
        height: 120, 
        borderRadius: 4
    }
});

export default class EpisodeRow extends React.Component {

    _getDate = (ms) => {
        const theDate = new Date(ms)
        const currentDate = new Date();

        const month = theDate.toLocaleString('en-us', { month: "long" })
        const day = theDate.getDate()
        const year = theDate.getFullYear();

        // Only show the year if it's past
        if (year == currentDate.getFullYear()) {
            return `${month} ${day}`;
        } else {
            return `${month} ${day}, ${year}`;
        }
    }

    render() {

        const { info, finished } = this.props
        const theDate = this._getDate(info.publishDate);

        return (
            <View opacity={finished ? 0.5 : 1}>
                <TouchableOpacity 
                style={{
                    borderBottomWidth: 1,
                    borderBottomColor: '#CCC',
                    padding: 20,
                    marginLeft: 10, 
                    marginRight: 10,
                    flexDirection: 'row',
                    height: 75
                }} onPress={this.props.onDetailPress}>
                    <View style={{
                        paddingRight: 10
                    }}>
                        <Text style={{
                            fontSize: 14,
                            fontWeight: "500",
                            color: '#666666'
                        }}>{theDate}</Text>
                        <Text style={{
                            fontSize: 16,
                            fontWeight: "600"
                        }} ellipsizeMode='tail' numberOfLines={1}>{info.title}</Text>
                    </View>
                    <TouchableOpacity style={{
                        paddingLeft: 20,
                        position: 'absolute',
                        right: 5,
                        alignItems: 'center',
                        alignContent: 'center',
                        justifyContent: 'center',
                        bottom: 0,
                        top: 0,
                        zIndex: 9
                    }} onPress={() => {
                        this.props.onPlayPress()
                    }}>
                        <Image 
                            source={require('../assets/play-gray.png')}
                            style={{
                                width: 20,
                                height: 20
                            }}
                        />
                    </TouchableOpacity>
                </TouchableOpacity>
            </View>
        );

    }
}