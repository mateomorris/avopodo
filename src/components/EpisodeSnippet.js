import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
    thumbnail: { 
        width: 120, 
        height: 120, 
        borderRadius: 4
    }
});

export default class EpisodeSnippet extends React.Component {

    state = {
        titleHeight : null
    }

    _handlePress = () => {
        this.props.onPress();
    }

    _normalizeDuration = (duration) => {
        let durationInHours = parseFloat(duration / (60 * 60)).toFixed(1)
        if (durationInHours < 1) {
            return `${durationInHours * 60}m`
        } else {
            return `${durationInHours}h`
        }
    }

    _getDate = (ms) => {
        const theDate = new Date(ms)
        const currentDate = new Date();

        const month = theDate.toLocaleString('en-us', { month: "long" })
        const day = theDate.getDate()
        const year = theDate.getFullYear();

        // Only show the year if it's past
        if (year == currentDate.getFullYear()) {
            return `${month} ${day}`;
        } else {
            return `${month} ${day}, ${year}`;
        }
    }

    render() {

        const { title, showImage, showImageHighRes, duration, description, publishDate, showColor } = this.props.data; 

        return (
            <TouchableOpacity style={{ flexDirection: 'row', marginBottom: 15, marginTop: 10, height: 130, overflow: 'hidden' }} onPress={() => {this.props.onPress()}}>
                <TouchableOpacity style={styles.container} onPress={() => { this.props.onThumbnailPress() }}>
                    <Image source={{uri: showImageHighRes || showImage}} style={[styles.thumbnail, {backgroundColor: showColor}]} />
                    <Image 
                        source={require('../assets/play.png')} 
                        style={{
                            height: 20,
                            width: 20,
                            position: 'absolute',
                            right: 5,
                            bottom: 5
                        }}
                    />
                    <View 
                        style={{ 
                            backgroundColor: 'black', 
                            borderBottomLeftRadius: 5,
                            borderTopRightRadius: 5,
                            paddingLeft: 5, 
                            paddingRight: 5, 
                            alignSelf: 'flex-start', 
                            marginLeft: 5 ,
                            position: 'absolute',
                            right: 0,
                            top: 0
                        }}>
                        <Text style={{ color: 'white', fontWeight: '900' }}>{this._normalizeDuration(duration)}</Text>
                    </View>
                </TouchableOpacity>
                <TouchableOpacity style={{ flex: 1, paddingLeft: 10 }} onPress={() => { this.props.onDetailPress() }}>
                    <View style={{ flexDirection: 'row', alignItems: 'center', paddingRight: 10}}>
                        <Text 
                            numberOfLines={2}
                            ellipsizeMode={'tail'}
                            style={{fontSize: 16, color: 'black', fontWeight: '700', flex: 1}} 
                            onLayout={(e) => {
                                this.setState({
                                    titleHeight : e.nativeEvent.layout.height
                                })
                        }}>{title}</Text>
                        {/* <View style={{ backgroundColor: 'black', borderRadius: 20, paddingLeft: 5, paddingRight: 5, alignSelf: 'flex-start', marginLeft: 5 }}>
                            <Text style={{ color: 'white', fontWeight: '900' }}>{this._normalizeDuration(duration)}</Text>
                        </View> */}
                    </View>
                    <View>
                        <Text 
                            style={{fontSize: 14, color: '#666666', fontWeight: '500'}}
                            numberOfLines={this.state.titleHeight == 20 ? 6 : 5} // Check if title is taking up more than one line
                            ellipsizeMode={'tail'}>
                            <Text style={{fontSize: 14, color: '#666666', fontWeight: '700'}}>{`${this._getDate(publishDate)}${ description ? ' | ' : ''}`}</Text>
                            {description}
                        </Text>
                    </View>
                </TouchableOpacity>
            </TouchableOpacity>
        );

    }
}