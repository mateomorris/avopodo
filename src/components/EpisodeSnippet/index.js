import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';
import SvgUri from 'react-native-svg-uri';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
    thumbnail: { 
        width: 120, 
        height: 120, 
        borderRadius: 5
    }
});

export default class EpisodeSnippet extends React.Component {

    state = {
        titleHeight : null,
        SVGs : {
            play : `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><title>button-play</title><path fill="black" d="M12,24A12,12,0,1,0,0,12,12.013,12.013,0,0,0,12,24Zm4.812-11.5a.939.939,0,0,1-.587.824L10.14,16.366a1.185,1.185,0,0,1-.531.133.919.919,0,0,1-.488-.136,1.032,1.032,0,0,1-.459-.911V9.546a.974.974,0,0,1,1.478-.914l6.085,3.043A.939.939,0,0,1,16.812,12.5Z"/></svg>`,
        }
    }

    _handlePress = () => {
        this.props.onPress();
    }

    _normalizeDuration = (duration) => {
        let durationInHours = parseFloat(duration / (60 * 60)).toFixed(1)
        if (durationInHours < 1) {
            return `${durationInHours * 60}m`
        } else {
            return `${durationInHours}h`
        }
    }

    _getDate = (ms) => {
        const theDate = new Date(ms)
        const currentDate = new Date();

        const month = theDate.toLocaleString('en-us', { month: "long" })
        const day = theDate.getDate()
        const year = theDate.getFullYear();

        // Only show the year if it's past
        if (year == currentDate.getFullYear()) {
            return `${month} ${day}`;
        } else {
            return `${month} ${day}, ${year}`;
        }
    }

    render() {

        const { title, showImage, showImageHighRes, duration, description, publishDate, showColor } = this.props.data; 

        return (
            <TouchableOpacity style={{ flexDirection: 'row', marginBottom: 10, marginTop: 5, height: 130, overflow: 'hidden' }} onPress={() => {this.props.onPress()}}>
                <TouchableOpacity style={styles.container} onPress={() => { this.props.onThumbnailPress() }}>
                    <Image source={{uri: showImageHighRes || showImage, cache: 'force-cache'}} style={[styles.thumbnail, {backgroundColor: showColor}]} />
                    {
                        <SvgUri style={{
                                height: 20,
                                width: 20,
                                position: 'absolute',
                                right: 5,
                                bottom: 5
                            }} width="20" height="20" svgXmlData={this.state.SVGs.play} fill={'#EEE'} fillAll={true}/>
                    }
                    <View 
                        style={{ 
                            backgroundColor: 'black', 
                            borderBottomLeftRadius: 5,
                            borderTopRightRadius: 5,
                            paddingLeft: 5, 
                            paddingRight: 5, 
                            alignSelf: 'flex-start', 
                            marginLeft: 5 ,
                            position: 'absolute',
                            right: 0,
                            top: 0
                        }}>
                        <Text style={{ color: 'white', fontWeight: '900' }}>{this._normalizeDuration(duration)}</Text>
                    </View>
                </TouchableOpacity>
                <TouchableOpacity style={{ flex: 1, paddingLeft: 10 }} onPress={() => { this.props.onDetailPress() }}>
                    <View style={{ flexDirection: 'row', alignItems: 'center', paddingRight: 10}}>
                        <Text 
                            numberOfLines={2}
                            ellipsizeMode={'tail'}
                            style={{fontSize: 16, color: 'black', fontWeight: '700', flex: 1}} 
                            onLayout={(e) => {
                                this.setState({
                                    titleHeight : e.nativeEvent.layout.height
                                })
                        }}>{title}</Text>
                        {/* <View style={{ backgroundColor: 'black', borderRadius: 20, paddingLeft: 5, paddingRight: 5, alignSelf: 'flex-start', marginLeft: 5 }}>
                            <Text style={{ color: 'white', fontWeight: '900' }}>{this._normalizeDuration(duration)}</Text>
                        </View> */}
                    </View>
                    <View>
                        <Text 
                            style={{fontSize: 14, color: '#666666', fontWeight: '500'}}
                            numberOfLines={this.state.titleHeight == 20 ? 6 : 5} // Check if title is taking up more than one line
                            ellipsizeMode={'tail'}>
                            <Text style={{fontSize: 14, color: '#666666', fontWeight: '700'}}>{`${this._getDate(publishDate)}${ description ? ' | ' : ''}`}</Text>
                            {description}
                        </Text>
                    </View>
                </TouchableOpacity>
            </TouchableOpacity>
        );

    }
}