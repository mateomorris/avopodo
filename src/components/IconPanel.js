import React from 'react';
import { StyleSheet, Text, View, Image, Dimensions, TouchableOpacity, Platform } from 'react-native';
import SvgUri from 'react-native-svg-uri';

const styles = StyleSheet.create({

});

export default class IconPanel extends React.Component {

    state = {
        selectedIcon : this.props.currentIcon || null
    }

    render() {

        let iconsWidth = Dimensions.get('window').width / 4 - 40
        let { icons } = this.props

        return (
            <View style={{
                flexWrap: 'wrap',
                flexDirection: 'row',
                justifyContent: 'flex-start'
            }}>
                {
                    Object.keys(icons).map((icon) => {
                        return (
                            <View opacity={this.state.selectedIcon == icon ? 0.5 : 1}>
                                <TouchableOpacity style={{
                                    margin: 10,
                                    width: iconsWidth,
                                    height: iconsWidth,
                                    padding: 10
                                }} onPress={() => {
                                    this.props.iconSelected(icon)
                                    this.setState({
                                        selectedIcon : icon
                                    })
                                }}>
                                    {/* {
                                        Platform.OS == 'ios' &&
                                        <SvgUri width="100%" height="100%" source={icons[icon]} fill={'white'} fillAll={true}/>
                                    } */}
                                </TouchableOpacity>
                            </View>
                        )
                    })
                }
            </View>
        );

    }
}