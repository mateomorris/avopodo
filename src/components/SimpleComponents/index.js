import React from 'react';
import { View, Dimensions, Text } from 'react-native';
import { DotIndicator } from 'react-native-indicators';

export const LoadingIndicator = () => {
    return (
        <View style={{ height: Dimensions.get('window').height - 200 }}>
            <DotIndicator color='gray' />
        </View>
    )
}

export const SmallLoadingIndicator = () => {
    return (
        <View style={{ height: 100 }}>
            <DotIndicator color='gray' />
        </View>
    )
}