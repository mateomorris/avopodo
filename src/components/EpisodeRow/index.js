import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity, Alert, ImageBackground } from 'react-native';
import * as Progress from 'react-native-progress';
import SvgUri from 'react-native-svg-uri';

import { CircleButton } from 'components/Button';
import icons from 'assets/generalIcons';

const styles = StyleSheet.create({
    container: {
        margin: 5, 
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 0
        },
        shadowRadius: 2,
        shadowOpacity: 0.75
    },
    thumbnail: { 
        width: 120, 
        height: 120, 
        borderRadius: 4
    }
});

export default class EpisodeRow extends React.Component {

    _getDate = (ms) => {
        const theDate = new Date(ms)
        const currentDate = new Date();

        const month = theDate.toLocaleString('en-us', { month: "long" })
        const day = theDate.getDate()
        const year = theDate.getFullYear();

        // Only show the year if it's past
        if (year == currentDate.getFullYear()) {
            return `${month} ${day}`;
        } else {
            return `${month} ${day}, ${year}`;
        }
    }

    render() {

        const { info, finished, buttonColor, playing } = this.props
        const theDate = this._getDate(info.publishDate);

        return (
            <View opacity={finished ? 0.5 : 1}>
                <TouchableOpacity 
                style={{
                    borderBottomWidth: 1,
                    borderBottomColor: '#CCC',
                    marginLeft: 10,
                    padding: 20,
                    paddingLeft: 10,
                    paddingRight: 10,
                    // marginLeft: 10, 
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                    alignItems: 'center',
                    width: '100%',
                    height: 120
                }} onPress={this.props.onDetailPress}>
                    <View style={{
                        paddingRight: 45
                    }}>
                        <Text style={{
                            fontSize: 14,
                            fontWeight: "500",
                            color: '#666666'
                        }}>{theDate}</Text>
                        <Text style={{
                            fontSize: 16,
                            fontWeight: "600",
                            color: '#333'
                        }} ellipsizeMode='tail' numberOfLines={1}>{info.title}</Text>
                        <Text style={{
                            fontSize: 14,
                            fontWeight: "500",
                            color: '#666666'
                        }} ellipsizeMode='tail' numberOfLines={3}>{info.description}</Text>
                    </View>
                    <View style={{
                        paddingLeft: 10,
                        paddingRight: 25,
                        position: 'absolute',
                        right: 0,
                        alignItems: 'center',
                        alignContent: 'center',
                        justifyContent: 'center',
                        bottom: 0,
                        top: 0,
                        zIndex: 9,
                    }}>
                        <CircleButton 
                            onPress={() => {
                                this.props.onPlayPress()
                            }}
                            icon={playing ? icons.pause : icons.play}
                            size={20}
                            spinner={false}
                            color={buttonColor}
                        />
                    </View>
                </TouchableOpacity>
            </View>
        );

    }
}