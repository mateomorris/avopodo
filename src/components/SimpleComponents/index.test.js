import React from 'react';
import { Text } from 'react-native';
import { LoadingIndicator, SmallLoadingIndicator } from 'components/SimpleComponents';
import renderer from 'react-test-renderer';

jest.useFakeTimers();
jest.mock('NativeAnimatedHelper');

it('renders LoadingIndicator correctly', () => {
  const tree = renderer
    .create(<LoadingIndicator></LoadingIndicator>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders SmallLoadingIndicator correctly', () => {
  const tree = renderer
    .create(<SmallLoadingIndicator></SmallLoadingIndicator>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});