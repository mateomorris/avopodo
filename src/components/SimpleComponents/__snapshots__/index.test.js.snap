// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoadingIndicator correctly 1`] = `
<View
  style={
    Object {
      "height": 1134,
    }
  }
>
  <View
    animating={true}
    animationDuration={1200}
    animationEasing={[Function]}
    color="gray"
    interaction={true}
    renderComponent={[Function]}
    size={16}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
  </View>
</View>
`;

exports[`renders SmallLoadingIndicator correctly 1`] = `
<View
  style={
    Object {
      "height": 100,
    }
  }
>
  <View
    animating={true}
    animationDuration={1200}
    animationEasing={[Function]}
    color="gray"
    interaction={true}
    renderComponent={[Function]}
    size={16}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "borderRadius": 8,
          "height": 16,
          "margin": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 16,
        }
      }
    />
  </View>
</View>
`;
